### DISCUSSION TITLE
High availability in the precense of node outages

### DISCUSSION THREAD

**Discussion by ecksun** [cloudnative-pg/cloudnative-pg#3501] [Q&A] [answered]
We are running a 3-node kubernetes cluster with CNPG. Our postgres cluster has 3 instances in total.

The scenario I'm not sure how to handle is when the operator and the primary cluster instance is both running on the same node and that node goes offline (say because of a power failure).

It takes kubernetes about 5 minutes to detect the failure and reschedule the operator and until such time the entire database cluster is unavailable.

Is it possible to run multiple instances of the operator or possibly use tolerations with `node.kubernetes.io/not-ready` or `node.kubernetes.io/unreachable` on the operator to get it to reschedule quicker?

**Comment by gbartolini** [comment, marked as answer]
Yes, it is possible. See https://cloudnative-pg.io/documentation/current/installation_upgrade/#details-about-the-deployment.

You might find useful also https://cloudnative-pg.io/documentation/current/kubectl-plugin/#generation-of-installation-manifests (--replicas)

**Reply by ecksun** [reply]
That is very useful, thank you very much :)
